src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,284 @@
+<template>
+  <div>
+    <div>
+      <label :id="'label-' + id">{{ selectLabel }} <span v-if=required aria-label='Required'>*</span></label>
+    </div>
+    <div
+      :class="dropdownClass"
+      :aria-expanded="opened"
+      :aria-owns="id"
+      :aria-required="required"
+      :aria-describedby="'error' + id"
+      :aria-labelledby="'label-' + id"
+      :id="'dropdown-' + id"
+      @keyup.space.prevent.stop="toggleList"
+      @click="toggleList"
+      tabindex="0"
+      v-model="selectLabel"
+      aria-haspopup="listbox"
+      >{{ selectedOption ? selectedOption.text : placeholder }}
+      <span class="dropdown-arrow"><i class="fa fa-caret-down" aria-hidden="true"></i></span>
+    </div>
+    <ul
+      :class="ulClass"
+      :id="id"
+      role="listbox"
+    >
+      <li
+        v-if=opened
+        v-for="option in selectOptions"
+        :aria-selected="selectedOption && selectedOption.id === option.id"
+        :id="'option-' + option.id"
+        :key="'selectOption' + option.id"
+        :value=option.id
+        @click="selectOption(option)"
+        @keyup.up.prevent="upHandler($event)"
+        @keyup.down.prevent="downHandler($event)"
+        @keyup.enter.prevent="selectOption(option)"
+        @keyup.space.prevent.stop="selectOption(option)"
+        class="options"
+        role="option"
+        tabindex='-1'
+      > {{ option.text }}
+      </li>
+    </ul>
+    <span v-show="selectErrors.has('selectedOption')" class="error" :id="'error-' + id">{{ selectErrors.first('selectedOption') }}</span>
+  </div>
+</template>
+
+<script>
+
+import { Validator } from 'vee-validate'
+
+const options = []
+
+export default {
+  name: 'b-select',
+  validator: null,
+  props: {
+    id: {
+      type: String,
+      required: true
+    },
+    selectOptions: {
+      type: Array
+    },
+    required: {
+      type: Boolean,
+      required: true
+    },
+    selectLabel: {
+      type: String,
+      required: true
+    },
+    initialValue: {
+      type: Object,
+      required: false
+    },
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Choose Option'
+    }
+  },
+
+  methods: {
+
+    /**
+     * Shows and hides line options for the select list.
+     */
+
```

Comment:
Generally anti-whitespace between comments and the code they describe. Other thoughts?